rules/macos/execution_curl_spawned_from_installer_package.toml

```
@@ -0,0 +1,71 @@
+[metadata]
+creation_date = "2021/02/23"
+maturity = "production"
+updated_date = "2021/02/23"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule identifies when the built in macOS Installer program is used to install a package that spawns a shell and runs
+the cURL command.
+"""
+false_positives = [
+    "Custom organization-specific macOS packages that use .pkg files to run cURL could trigger this rule.",
+]
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
```

This _should_ be compatible with auditbeat too
```suggestion
index = ["auditbeat-*", "logs-endpoint.events.*"]
```

---

This should work with streamed events in the `"endgame-*"` index as well.

---

Auditbeat appears to be missing:
- process.parent.executable
- process.command_line

I don't have access to any `endgame` indexes. Do those fields exist there?

---

@dcode did you have `endgame-*` indices you could search to see if those fields are present?

---

so...sorry for this, but I've abstracted this query to be a bit more inclusive...namely, find any time a child process of the installer initiates a network connection. The Endgame data isn't compatible, and I'm not sure if auditbeat has the necessary events, but I'll give it a look.

Here's the ECS version. The cidrMatch isn't my favorite, but was trying to rule out an event that happens against the local network, and I'm not quite sure why just yet. I also tried  `network where destination.geo.country_iso_code != null`, but the depends on geo enrichment at ingest time.

```
sequence by process.entity_id
  [ process where host.os.family == "macos" and (process.parent.executable == "/usr/sbin/installer" or process.parent.executable == "/System/Library/CoreServices/Installer.app/Contents/MacOS/Installer") ]
  [ network where not cidrMatch(destination.ip, "192.168.0.0/16", "10.0.0.0/8", "172.16.0.0/12", "224.0.0.0/8") ]
```

Here's the Endgame version. The indices I queried against didn't have `process.entity_id`
```
sequence by labels.endpoint_id, process.pid
  [ process where host.os.family == "macos" and (process.parent.executable == "/usr/sbin/installer" or process.parent.executable == "/System/Library/CoreServices/Installer.app/Contents/MacOS/Installer") ]
  [ network where not cidrMatch(destination.ip, "192.168.0.0/16", "10.0.0.0/8", "172.16.0.0/12", "224.0.0.0/4") ]
```

---

In this scenario I would imagine an internal network connection would be just as odd (if not more in some instances) as an external connection. Perhaps we can add notes to `false_positive` about potentially excluding such network activity, dependent on what their respective environments look like

---

I dug deeper to see if I could find something equivalent in auditbeat, but auditbeat doesn't have socket logging support on macOS yet :-/. I also completely agree that an internal network connection is odd. I _think_ that connection was `curl` resolving the IP from the gateway (also nameserver) via mDNSResponder and may be a bug in how the data is recorded. I don't see any other way from the contents of SILVER SPARROW itself where it would make a connection at all.

It'd be better if we could say as part of the sequence: one or more network connections and capture them all, but I don't think we can do that.

---

Got it...use the sequence as-is and create a timeline based on `process.entity_id`

---

Looks like we've got a good EQL rule.